dev/src/components/Members.tsx

```diff
@@ -67,7 +68,7 @@ const Members = (): JSX.Element => {
             data.push(edit ? (
                 <div>
                     <i className="badge badge-primary clickable"
-                        onClick={() => actions.updateEnrollment({ enrollment: enrollment, status: isTeacher(enrollment) ? Enrollment.UserStatus.STUDENT : Enrollment.UserStatus.TEACHER })}>
+                        onClick={() => { actions.updateEnrollment({ enrollment: enrollment, status: isTeacher(enrollment) ? Enrollment.UserStatus.STUDENT : Enrollment.UserStatus.TEACHER }); setPromote(!promote) }}>
```

Comment:
Currently neither `accept` nor `promote` are used to control which text is displayed. 

Note also that the `accept` and `promote` states are function scoped. `members` is a nested function `.map(enrollment => { ... })`, which returns an array of rows representing individual members in the table. 

The issue that arises if `setPromote(!promote)` is invoked is that it will affect *all* rows. Flipping `promote` from `true -> false` will be reflected in all rows, rather than only the one clicked.

You can avoid this by moving the `approve/reject` and `promote/demote` buttons into two separate components having their own local state.

You will also have to handle the case where updating the enrollment status is unsuccessful.

However, I suggest handling the issue in the `updateEnrollment` action:

```ts
        if (success(response)) {
            // If successful, update enrollment in state with new status
            if (status == Enrollment.UserStatus.NONE) {
                // If the enrollment is rejected, remove it from state
                state.courseEnrollments[state.activeCourse] = state.courseEnrollments[state.activeCourse]?.filter(s => s.getId() != enrollment.getId())
            } else {
                const enrollment = state.courseEnrollments[state.activeCourse]?.find(s => s.getId() == temp.getId())
                if (enrollment) {
                    enrollment.setStatus(status)
                }
            }
        }
```

Line 7-10.

The reason the text does not update is that `state.courseEnrollments[courseID]` is wrapped in `json(...)` in `sortEnrollments(...)`.
The issue with `json(...)` is that is removes a proxy wrapper from our state object, which is used to tell Overmind when to trigger a re-render of components. 

Updating the enrollment passed in to `updateEnrollment` does change the state, but it will not trigger a re-render. Manually triggering a re-render of the `Members` component will update the text to reflect the desired value.
The change on line 7-10 retrieves a new reference to the enrollment, wrapped in a proxy. Modifying the status of this enrollment will automatically trigger a re-render with the correct text.